packages/@lwc/engine-core/src/framework/restrictions.ts

```diff
@@ -91,31 +91,31 @@ function getNodeRestrictionsDescriptors(
     return {
         appendChild: generateDataDescriptor({
             value(this: Node, aChild: Node) {
-                if (this instanceof Element && isFalse(options.isPortal)) {
```

Comment:
@caridy do you recall why the `Element` check was needed here? It was introduced with the `lwc:dom="manual"` directive: https://github.com/salesforce/lwc/commit/bd64274a78a2048cca5ed29edce20505613d7936#diff-8d59e8cac3f4cb87581a75c21f6d97e7R41-R44. We also don't have any specific test coverage for this either.

Reply:
Could be due to this:
The restrictions routine `getNodeRestrictionsDescriptors` is invoked for any Node and the portal restrictions are only for Elements.
Currently this method is invoked for the host, shadowRoot and all elements. Host and shadowRoot do not need these restrictions.

Reply:
I don't remember exactly why, but @ravijayaramappa 's assumptions sound about right. The shadowRoot/fragments are the problem here. We can chat about it.

Reply:
Good point @ravijayaramappa. I created another PR to work around this issue without relying on `instanceof`: https://github.com/salesforce/lwc/pull/1934